jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DelayedHandler.java

```diff
@@ -167,75 +169,133 @@ protected Callback getCallback()
         }
 
         protected void process()
+        {
+            process(getRequest(), getResponse(), getCallback());
+        }
+
+        protected boolean process(Request request, Response response, Callback callback)
         {
             try
             {
-                if (!getHandler().handle(getRequest(), getResponse(), getCallback()))
-                    Response.writeError(getRequest(), getResponse(), getCallback(), HttpStatus.NOT_FOUND_404);
+                if (getHandler().handle(request, response, callback))
+                    return true;
+
+                Response.writeError(getRequest(), getResponse(), getCallback(), HttpStatus.NOT_FOUND_404);
             }
             catch (Throwable t)
             {
                 Response.writeError(getRequest(), getResponse(), getCallback(), t);
             }
+            return false;
         }
 
         protected abstract void delay() throws Exception;
     }
 
-    protected static class UntilContentDelayedProcess extends DelayedProcess
+    /**
+     * Delay dispatch until all content or 75% of an input buffer is received.
+     */
+    protected static class UntilContentDelayedProcess extends DelayedProcess implements Runnable
     {
+        private final Deque<Content.Chunk> _chunks = new ArrayDeque<>();
+        private final int _maxBuffered;
+        private int _size;
+
         public UntilContentDelayedProcess(Handler handler, Request request, Response response, Callback callback)
         {
             super(handler, request, response, callback);
+            _maxBuffered = 3 * request.getConnectionMetaData().getConnector().getConnectionFactory(HttpConnectionFactory.class).getInputBufferSize() / 4;
```

Comment:
Same as above: how do you justify 75%?

Reply:
I'm now using a guessed capacity, which is initialized at the buffer size, then reduced 8 bytes for each chunk (guessed framing overhead) and the chunk size itself.